src/app/InteractionModelEngine.cpp

```diff
@@ -172,8 +172,8 @@ CHIP_ERROR InteractionModelEngine::NewWriteClient(WriteClient ** const apWriteCl
             continue;
         }
 
-        ReturnErrorOnFailure(writeClient.Init(mpExchangeMgr, mpDelegate));
-        *apWriteClient = &writeClient;
+        ReturnErrorOnFailure(writeClient.Init(mpExchangeMgr, mpDelegate, aApplicationIdentifier));
```

Comment:
Consider to replace ReturnErrorOnFailure with ReturnLogErrorOnFailure when you feel you need to log error log error in initial error point